pkg/storegateway/indexheader/index/postings.go

```diff
@@ -51,11 +51,15 @@ type PostingOffsetTableV1 struct {
 	postings map[string]map[string]index.Range
 }
 
-func NewPostingOffsetTable(factory *streamencoding.DecbufFactory, tableOffset int, indexVersion int, indexLastPostingListEndBound uint64, postingOffsetsInMemSampling int) (PostingOffsetTable, error) {
+func NewPostingOffsetTable(factory *streamencoding.DecbufFactory, tableOffset int, indexVersion int, indexLastPostingListEndBound uint64, postingOffsetsInMemSampling int, doChecksum bool) (PostingOffsetTable, error) {
 	if indexVersion == index.FormatV1 {
 		return newV1PostingOffsetTable(factory, tableOffset, indexLastPostingListEndBound)
 	} else if indexVersion == index.FormatV2 {
-		return newV2PostingOffsetTable(factory, tableOffset, indexLastPostingListEndBound, postingOffsetsInMemSampling)
+		tab := castagnoliTable
+		if !doChecksum {
+			tab = nil
+		}
+		return newV2PostingOffsetTable(factory, tab, tableOffset, indexLastPostingListEndBound, postingOffsetsInMemSampling)
```

Comment:
[nit] Have you considered push down `doChecksum` to `newV2PostingOffsetTable()`?